### ISSUE TITLE
Fix operations effects when signer changes.

### ISSUE THREAD

**Issue by abuiles** [stellar/go#2136]
Every time there is a signer change on an account with multiple signers, we are incorrectly reporting a signer updated effect on the other unaffected signers.

For example, the following operation https://horizon.stellar.org/operations/71491809406156801/effects  is showing three effects, however, if we look at the XDR representation for the operation.

```go
      { sourceAccount = GCHHFZV3NDPRHPKK2OVJI36CNLOOJUY5BX7HQD4OMC2NHCLQ6YHYF5WC,
        body = {
          type = SET_OPTIONS,
          setOptionsOp = {
            inflationDest = NULL,
            clearFlags = NULL,
            setFlags = NULL,
            masterWeight = NULL,
            lowThreshold = NULL,
            medThreshold = NULL,
            highThreshold = NULL,
            homeDomain = NULL,
            signer = {
              key = {
                type = SIGNER_KEY_TYPE_ED25519,
                ed25519 = ed43b19ae7eb0e1a3f28d3ffe9e3aa27e5a4509f4b1548910f6d26b377d94525
              },
              weight = 0
            }
          }
        } },
```

It is only removing one signer and the weight for the other signers is kept unchanged.

<details>
  <summary>See full Tx Envelope</summary>
  
```go
TransactionEnvelope = {
  tx = {
    sourceAccount = GDW6N6JLDMZJJGMYKZAHWDFCJXX3QNP6EEMTJ5PSHKTJBAVERLXNOJ32,
    fee = 400,
    seqNum = 70990698391797762,
    timeBounds = {
      minTime = 1520266282,
      maxTime = 0
    },
    memo = {
      type = MEMO_NONE
    },
    operations = [
      { sourceAccount = GCHHFZV3NDPRHPKK2OVJI36CNLOOJUY5BX7HQD4OMC2NHCLQ6YHYF5WC,
        body = {
          type = SET_OPTIONS,
          setOptionsOp = {
            inflationDest = NULL,
            clearFlags = NULL,
            setFlags = NULL,
            masterWeight = NULL,
            lowThreshold = NULL,
            medThreshold = NULL,
            highThreshold = NULL,
            homeDomain = NULL,
            signer = {
              key = {
                type = SIGNER_KEY_TYPE_ED25519,
                ed25519 = ed43b19ae7eb0e1a3f28d3ffe9e3aa27e5a4509f4b1548910f6d26b377d94525
              },
              weight = 0
            }
          }
        } },
      { sourceAccount = GCHHFZV3NDPRHPKK2OVJI36CNLOOJUY5BX7HQD4OMC2NHCLQ6YHYF5WC,
        body = {
          type = SET_OPTIONS,
          setOptionsOp = {
            inflationDest = NULL,
            clearFlags = NULL,
            setFlags = NULL,
            masterWeight = NULL,
            lowThreshold = NULL,
            medThreshold = NULL,
            highThreshold = NULL,
            homeDomain = NULL,
            signer = {
              key = {
                type = SIGNER_KEY_TYPE_HASH_X,
                hashX = 762329959e01e9716a7008835bc3944361d4b87d76f741ea60a167f28dc99107
              },
              weight = 0
            }
          }
        } },
      { sourceAccount = GCHHFZV3NDPRHPKK2OVJI36CNLOOJUY5BX7HQD4OMC2NHCLQ6YHYF5WC,
        body = {
          type = SET_OPTIONS,
          setOptionsOp = {
            inflationDest = NULL,
            clearFlags = NULL,
            setFlags = NULL,
            masterWeight = NULL,
            lowThreshold = NULL,
            medThreshold = NULL,
            highThreshold = NULL,
            homeDomain = NULL,
            signer = {
              key = {
                type = SIGNER_KEY_TYPE_ED25519,
                ed25519 = ede6f92b1b3294999856407b0ca24defb835fe211934f5f23aa69082a48aeed7
              },
              weight = 1
            }
          }
        } },
      { sourceAccount = GCHHFZV3NDPRHPKK2OVJI36CNLOOJUY5BX7HQD4OMC2NHCLQ6YHYF5WC,
        body = {
          type = SET_OPTIONS,
          setOptionsOp = {
            inflationDest = NULL,
            clearFlags = NULL,
            setFlags = NULL,
            masterWeight = NULL,
            lowThreshold = 1,
            medThreshold = 1,
            highThreshold = 1,
            homeDomain = NULL,
            signer = NULL
          }
        } }
    ],
    ext = {
      v = 0
    }
  },
  signatures = [
  ]
}
```
</details>

The root of this problem is on the following line https://github.com/stellar/go/blob/2e9e66756e0822e0864934008d66ad7c5b3edd20/services/horizon/internal/expingest/processors/effects_processor.go#L487-L499

When looking at the `before` and `after`  state, we add a delete event if the key is not present on the new state; otherwise, we add an update event.

To fix this, we need to add an extra check on the `weight`, and if it's different, then it is an update; otherwise we should not add an effect.

**Comment by abuiles** [comment]
From @MonsieurNicolas 
```
@jrice What do you think of removing the duplicate debit/remove effects in an account merge attack (see last 4 effects of https://horizon.stellar.org/accounts/GBSSDVJTXF6XJUN5EWCUXDMCGYGNCIDTPL6G6WM6FTG3G4GDGDKU7Z2T/effects )
The last two debit/remove effects on the already merged account actually never affected the state, so it seems more correct to omit them. Only the account_credit effect on the receiving account is applied twice.
Otherwise, summing all seen amounts in effects of an account could produce negative balances.
```

**Comment by abuiles** [comment]
Fixed in #2349
